### DISCUSSION TITLE
Telegram: Could not upload image with chatid ' '

### DISCUSSION THREAD

**Discussion by wilddoktor** [VorlonCD/bi-aidetection#310] [Q&A] [answered]
Hi all; I'm running AI Tools v 2.0.37.7608.

I decided to try checking the "send to telegram" box tonight. I got a bot setup, and messages sending properly, and got the correct token and chatid, and plugged them into AI Tool - Settings.

But when I do a test on the camera (Cameras - Actions - Settings - Test), I get the following error:

"Could not upload image g:\AlertImages\hd1.20220112_221853338_AITOOLTEST_80409.1294115.jpg to Telegram with chatid '': Invalid format. A valid token looks like "1234567:4TT8bAc8GHUspu3ERYn-KGcvsvGB9u_n4ddy". ; Parameter name: token [ArgumentException] Mod: <TelegramUpload>d__44 Line:836:33"

Unsure why the error is saying my chatid is blank (it isn't) and the token isn't valid.

Any help would be great! Thanks!

**Comment by VorlonCD** [comment, marked as answer]
First, try the latest:  2.1.6:
https://github.com/VorlonCD/bi-aidetection/tree/master/src/UI/Installer

Also, DOES your token look like the format in their example?

**Comment by wilddoktor** [comment]
Thanks, will give that a try after work today!

The token looks just like the format; number, : , alpha-numeric. Chat ID starts with a -.

**Comment by wilddoktor** [comment]
Oh, and I haven't updated vorlonCD in like a year; will I lose settings and have to re-input when I run the exe mentioned above? Thanks!

**Comment by VorlonCD** [comment]
you should not loose any settings.   The exe is an installer, not the actual aitool.

**Comment by wilddoktor** [comment]
Awesome, thanks for the quick reply!

**Comment by wilddoktor** [comment]
Thank you so much!! That did the trick for me; I'm getting alerts in Telegram now! Lots of cool new things to try out, but pretty intuitive. 

Again, thank you...and, great job!
